pkg/cmd/server/start/start_scheduler.go

```diff
@@ -0,0 +1,53 @@
+package start
+
+import (
+	"github.com/golang/glog"
+
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	schedulerapp "k8s.io/kubernetes/plugin/cmd/kube-scheduler/app"
+	scheduleroptions "k8s.io/kubernetes/plugin/cmd/kube-scheduler/app/options"
+	_ "k8s.io/kubernetes/plugin/pkg/scheduler/algorithmprovider"
+
+	cmdflags "github.com/openshift/origin/pkg/cmd/util/flags"
+)
+
+func newScheduler(kubeconfigFile, schedulerConfigFile string, cmdLineArgs map[string][]string) (*scheduleroptions.SchedulerServer, error) {
+	if cmdLineArgs == nil {
+		cmdLineArgs = map[string][]string{}
+	}
+	if len(cmdLineArgs["kubeconfig"]) == 0 {
+		cmdLineArgs["kubeconfig"] = []string{kubeconfigFile}
+	}
+	if len(cmdLineArgs["policy-config-file"]) == 0 {
+		cmdLineArgs["policy-config-file"] = []string{schedulerConfigFile}
+	}
+	// disable serving http since we didn't used to expose it
+	if len(cmdLineArgs["port"]) == 0 {
+		cmdLineArgs["port"] = []string{"-1"}
+	}
+
+	// resolve arguments
+	schedulerServer := scheduleroptions.NewSchedulerServer()
+	if err := cmdflags.Resolve(cmdLineArgs, schedulerServer.AddFlags); len(err) > 0 {
+		return nil, kerrors.NewAggregate(err)
+	}
+
+	return schedulerServer, nil
+}
+
+func runEmbeddedScheduler(kubeconfigFile, schedulerConfigFile string, cmdLineArgs map[string][]string) {
+	for {
+		// TODO we need a real identity for this.  Right now it's just using the loopback connection like it used to.
+		scheduler, err := newScheduler(kubeconfigFile, schedulerConfigFile, cmdLineArgs)
+		if err != nil {
+			glog.Error(err)
+			continue
```

Comment:
It seems to me that if there is any parsing error or anything, it will keep filling the logs. How continuing here is helpful?

Reply:
> It seems to me that if there is any parsing error or anything, it will keep filling the logs. How continuing here is helpful?

It keeps the entire process (including the apiserver) from dying.